configs/jsactions/rollup.config.js

```diff
@@ -0,0 +1,144 @@
+import { red, yellow } from "colors";
+import { mkdirSync, promises } from "fs";
+import { copy } from "fs-extra";
+import { basename, dirname, extname, join, relative } from "path";
+import clear from "rollup-plugin-clear";
+import command from "rollup-plugin-command";
+import { cp } from "shelljs";
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import typescript from "@rollup/plugin-typescript";
+import { collectDependencies } from "../../packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-dependencies";
+
+const cwd = process.cwd();
+
+export default async args => {
+    const jsActionTargetFolder = `javascriptsource/${args.configProject ?? "nativemobileresources"}/actions`;
+    const result = [];
+    const files = await findFiles();
+    const outDir = join(cwd, "dist");
+
+    files.forEach((file, i) => {
+        const fileInput = relative(cwd, file.path);
+        const [fileOutput] = file.name.split(".");
+        result.push({
+            input: fileInput,
+            output: {
+                format: "es",
+                file: join(outDir, `${fileOutput}.js`),
+                sourcemap: false
+            },
+            external: nativeExternal,
+            plugins: [
+                i === 0 ? clear({ targets: [outDir] }) : null,
+                collectDependencies({
+                    externals: nativeExternal,
+                    isJSAction: true,
+                    outputDir: outDir,
+                    shouldCopyNodeModules: true,
+                    shouldRemoveNodeModules: i === 0,
+                    widgetName: fileOutput
+                }),
+                nodeResolve({ preferBuiltins: false, mainFields: ["module", "browser", "main"] }),
+                typescript({
+                    noEmitOnError: false,
+                    sourceMap: false,
+                    inlineSources: false,
+                    target: "es2019",
+                    types: ["mendix-client", "big.js", "react-native"],
+                    allowSyntheticDefaultImports: true
+                }),
+                i === files.length - 1
+                    ? command([
+                          async () => {
+                              await Promise.all(
+                                  clientDependencies.map(async dependency => {
+                                      await copyJsModule(
+                                          dirname(require.resolve(`${dependency}/package.json`)),
+                                          join(join(outDir, "node_modules"), dependency)
```

Comment:
Won't this result in double-inclusion of modules that are used from widgets and js actions, like firebase?

Reply:
They will never be bundled together right?

Reply:
Yes, but if a project uses a js action that depend on firebase and widget that depend on firebase - we have a problem, since firebase is included twice. Or do i miss something?

Reply:
No, because if the lib is native dep it will be caught in line 74, if not then it will add them into the JSdependencies even if the isJSAction flag is turned on.